fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/contract/routing/impl/SerializerRegistryImpl.java

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package org.hyperledger.fabric.contract.routing.impl;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.hyperledger.fabric.Logger;
+import org.hyperledger.fabric.contract.annotation.Serializer;
+import org.hyperledger.fabric.contract.execution.SerializerInterface;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ScanResult;
+
+/**
+ * Registry to hold permit access to the routing definitions. This is the
+ * primary internal data structure to permit access to information about the
+ * contracts, and their transaction functions.
+ *
+ * Contracts are added, and processed. At runtime, this can then be accessed to
+ * locate a specific 'Route' that can be handed off to the ExecutionService
+ *
+ * Type T should 
+ * 
+ */
+public class SerializerRegistryImpl {
+    private static Logger logger = Logger.getLogger(SerializerRegistryImpl.class);
+
+    private Class<Serializer> annotationClass;
+
+    public SerializerRegistryImpl() {
+        this.annotationClass = Serializer.class;
+    }
+
+    // Could index these by name and or type.
+    private Map<String, SerializerInterface> contents = new HashMap<>();
+
+    /**
+     * Get a Serializer for the matching name, and the Target
+     */
+    public SerializerInterface getSerializer(String name, Serializer.TARGET target) {
```

Comment:
Name of what? Contract name?

Reply:
Fully qualified serialiser class name. Would be nice to include arguments in the JavaDoc.